src/integration-test/java/com/commercetools/sync/integration/externalsource/types/TypeSyncIT.java

```diff
@@ -0,0 +1,712 @@
+import io.sphere.sdk.types.EnumFieldType;
+import io.sphere.sdk.types.FieldDefinition;
+import io.sphere.sdk.types.LocalizedEnumFieldType;
+import io.sphere.sdk.types.ResourceTypeIdsSetBuilder;
+import io.sphere.sdk.types.StringFieldType;
+import io.sphere.sdk.types.Type;
+import io.sphere.sdk.types.TypeDraft;
+import io.sphere.sdk.types.TypeDraftBuilder;
+import io.sphere.sdk.types.commands.TypeUpdateCommand;
+import io.sphere.sdk.types.queries.TypeQuery;
+import io.sphere.sdk.utils.CompletableFutureUtils;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletionException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.commercetools.sync.integration.commons.utils.ITUtils.deleteTypes;
+import static com.commercetools.sync.integration.commons.utils.SphereClientUtils.CTP_TARGET_CLIENT;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.FIELD_DEFINITION_1;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.FIELD_DEFINITION_2;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.FIELD_DEFINITION_3;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.FIELD_DEFINITION_LABEL_1;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.FIELD_DEFINITION_NAME_1;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.TYPE_DESCRIPTION_1;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.TYPE_DESCRIPTION_2;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.TYPE_KEY_1;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.TYPE_KEY_2;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.TYPE_NAME_1;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.TYPE_NAME_2;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.getTypeByKey;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.populateSourceProject;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.populateTargetProject;
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+
+public class TypeSyncIT {
+
+    /**
+     * Deletes types from source and target CTP projects.
+     * Populates source and target CTP projects with test data.
+     */
+    @Before
+    public void setup() {
+        deleteTypes(CTP_TARGET_CLIENT);
+        populateSourceProject();
```

Comment:
Why do you need a source project if these tests are testing external source not CTP project?